AsyncDisplayKit/ASControlNode.mm

```diff
@@ -278,32 +275,28 @@ - (void)addTarget:(id)target action:(SEL)action forControlEvents:(ASControlNodeE
   _ASEnumerateControlEventsIncludedInMaskWithBlock(controlEventMask, ^
     (ASControlNodeEvent controlEvent)
     {
+      // Create new target action pair
+      ASControlTargetAction *targetAtion = [[ASControlTargetAction alloc] init];
+      targetAtion.action = action;
+      targetAtion.target = target;
+      
       // Do we already have an event table for this control event?
       id<NSCopying> eventKey = _ASControlNodeEventKeyForControlEvent(controlEvent);
-      NSMapTable *eventDispatchTable = _controlEventDispatchTable[eventKey];
-      // Create it if necessary.
-      if (!eventDispatchTable)
-      {
-        // Create the dispatch table for this event.
-        eventDispatchTable = [NSMapTable weakToStrongObjectsMapTable];
-        if (eventKey) {
-          [_controlEventDispatchTable setObject:eventDispatchTable forKey:eventKey];
-        }
+      NSMutableOrderedSet *eventTargetActionArray = _controlEventDispatchTable[eventKey];
+      
+      if (!eventTargetActionArray) {
+        eventTargetActionArray = [[NSMutableOrderedSet alloc] init];
       }
-
-      // Have we seen this target before for this event?
-      NSMutableSet *targetActions = [eventDispatchTable objectForKey:target];
-      if (!targetActions)
-      {
-        // Nope. Create an action set for it.
-        targetActions = [[NSMutableSet alloc] initWithCapacity:kASControlNodeActionDispatchTableInitialCapacity]; // enough to handle common types without re-hashing the dictionary when adding entries.
-        [eventDispatchTable setObject:targetActions forKey:target];
+      
+      // Remove any prior target-action pair for this event, as UIKit does.
+      [eventTargetActionArray removeObject:targetAtion];
+      
+      // Register the new target-action as the last one to be sent.
+      [eventTargetActionArray addObject:targetAtion];
```

Comment:
Now that we're manually removing any prior target-action, we can switch back to using `NSMutableArray` which is a little faster.